### PR TITLE
Fixing up coverity report flags from #1760 and #1761

### LINKED ISSUE #1760: 3 new coverity findings
@Evengard , can you please have a look ?  

```
*** CID 383118:    (OVERRUN)
/src/Cedar/Proto_PPP.c: 3593 in PPPProcessEAPTlsResponse()
3587                            for (i = 0; i < dataSize; i++)
3588                            {
3589                                    if (i > 0) printf(" ");
3590                                    Debug("%02X", dataBuffer[i]);
3591                            }
3592                            Debug("\n=======RECV EAP-TLS PACKET FIFO END=======\n");*/
>>>     CID 383118:    (OVERRUN)
>>>     Calling "WriteFifo" with "dataBuffer" and "dataSize" is suspicious because of the very large index, 4294967292. The index may be due to a negative parameter being interpreted as unsigned.
3593                            WriteFifo(p->Eap_TlsCtx.SslPipe->RawIn->SendFifo, dataBuffer, dataSize);
3594                            syncOk = SyncSslPipe(p->Eap_TlsCtx.SslPipe);
3595     
3596                            // Delete the cached buffer after we fed it into the pipe
3597                            if (p->Eap_TlsCtx.CachedBufferRecv != NULL)
3598                            {
/src/Cedar/Proto_PPP.c: 3553 in PPPProcessEAPTlsResponse()
3547                                    p->Eap_TlsCtx.CachedBufferRecv = ZeroMalloc(MAX(dataSize, PPP_MRU_MAX * 10)); // 10 MRUs should be enough
3548                                    p->Eap_TlsCtx.CachedBufferRecvPntr = p->Eap_TlsCtx.CachedBufferRecv;
3549                            }
3550                            sizeLeft = GetMemSize(p->Eap_TlsCtx.CachedBufferRecv);
3551                            sizeLeft -= (UINT)(p->Eap_TlsCtx.CachedBufferRecvPntr - p->Eap_TlsCtx.CachedBufferRecv);
3552     
>>>     CID 383118:    (OVERRUN)
>>>     Calling "Copy" with "dataBuffer" and "(sizeLeft >= dataSize) ? dataSize : sizeLeft" is suspicious because of the very large index, 4294967292. The index may be due to a negative parameter being interpreted as unsigned.
3553                            Copy(p->Eap_TlsCtx.CachedBufferRecvPntr, dataBuffer, MIN(sizeLeft, dataSize));
3554     
3555                            p->Eap_TlsCtx.CachedBufferRecvPntr += MIN(sizeLeft, dataSize);
3556                    }
3557     
3558                    // If we got a cached buffer, we should feed the FIFOs via it

** CID 383117:  Memory - corruptions  (OVERRUN)


________________________________________________________________________________________________________
*** CID 383117:  Memory - corruptions  (OVERRUN)
/src/Cedar/Proto_PPP.c: 3553 in PPPProcessEAPTlsResponse()
3547                                    p->Eap_TlsCtx.CachedBufferRecv = ZeroMalloc(MAX(dataSize, PPP_MRU_MAX * 10)); // 10 MRUs should be enough
3548                                    p->Eap_TlsCtx.CachedBufferRecvPntr = p->Eap_TlsCtx.CachedBufferRecv;
3549                            }
3550                            sizeLeft = GetMemSize(p->Eap_TlsCtx.CachedBufferRecv);
3551                            sizeLeft -= (UINT)(p->Eap_TlsCtx.CachedBufferRecvPntr - p->Eap_TlsCtx.CachedBufferRecv);
3552     
>>>     CID 383117:  Memory - corruptions  (OVERRUN)
>>>     Calling "Copy" with "p->Eap_TlsCtx.CachedBufferRecvPntr" and "(sizeLeft >= dataSize) ? dataSize : sizeLeft" is suspicious because of the very large index, 4294967292. The index may be due to a negative parameter being interpreted as unsigned.
3553                            Copy(p->Eap_TlsCtx.CachedBufferRecvPntr, dataBuffer, MIN(sizeLeft, dataSize));
3554     
3555                            p->Eap_TlsCtx.CachedBufferRecvPntr += MIN(sizeLeft, dataSize);
3556                    }
3557     
3558                    // If we got a cached buffer, we should feed the FIFOs via it

** CID 383116:  Uninitialized variables  (UNINIT)
/src/Cedar/Proto_PPP.c: 1339 in PPPProcessEAPResponsePacket()


________________________________________________________________________________________________________
*** CID 383116:  Uninitialized variables  (UNINIT)
/src/Cedar/Proto_PPP.c: 1339 in PPPProcessEAPResponsePacket()
1333                                            return false;
1334                                    }
1335                                    break;
1336                            }
1337     
1338                            // Select EAP method based on auth type
>>>     CID 383116:  Uninitialized variables  (UNINIT)
>>>     Using uninitialized value "authtype".
1339                            switch (authtype)
1340                            {
1341                            case AUTHTYPE_RADIUS:
1342                                    // Create EAP client if needed
1343                                    if (p->MsChapV2_UseDoubleMsChapV2 && p->EapClient == NULL)
1344                                    {

```

### PR DESCRIPTION
**Pull request by Evengard**
Should fix #1760 and #1761.
Some of them were ignored as being just redundant checks (defensive programming, may help if code changes) or bogus (coverity apparently doesn't understand the SoftEther memtags).

### CODE REVIEW

**Comment by chipitsine** [comment]
@domosekai , can you please review (maybe some issues introduced while fixing code analysis findings ...)
thanks!

**Comment by domosekai** [comment]
Sorry, I wish but I lack the information needed as well as the whole picture. 
Can we make the coverity result publicly accessible?

**Comment by Evengard** [comment]
I've actually managed to get a read-only access via https://scan.coverity.com/projects/softethervpn-softethervpn/view_defects - you can try as well.

**Comment by domosekai** [comment]
I am not asking for myself.

**Comment by chipitsine** [comment]
Please join the project, I will approve

**Comment by chipitsine** [comment]
Even it is set up to show defect to all users, seems it only applies to
users who joined the project

**Comment by chipitsine** [comment]
Even it is set up to show defect to all users, seems it only applies to
users who joined the project


image

File: src/Cedar/Proto_PPP.c [hunk 1]
```diff
@@ -2322,8 +2323,7 @@ bool PPPRejectLCPOptionsEx(PPP_SESSION *p, PPP_PACKET *pp, bool simulate)
 		return false;
 	}
 
-	PPPSendPacketAndFree(p, ret);
-	return true;
+	return PPPSendPacketAndFree(p, ret);
```
Comment (domosekai) [inline src/Cedar/Proto_PPP.c:2326]: return true implies that the packet should be rejected. I would leave it.

File: src/Cedar/Proto_PPP.c [hunk 2]
```diff
@@ -2375,8 +2375,7 @@ bool PPPNackLCPOptionsEx(PPP_SESSION *p, PPP_PACKET *pp, bool simulate)
...
```
Comment (domosekai) [inline src/Cedar/Proto_PPP.c:2378]: same reason. I would choose to leave it.

File: src/Cedar/Proto_PPP.c [hunk 3]
```diff
@@ -1278,6 +1278,7 @@ bool PPPProcessEAPResponsePacket(PPP_SESSION *p, PPP_PACKET *pp, PPP_PACKET *req
 		switch (eap_packet->Type)
 		{
 		case PPP_EAP_TYPE_IDENTITY:
+			p->Eap_MatchUserByCert = false;
```
Comment (domosekai) [inline src/Cedar/Proto_PPP.c:1281]: The fix is correct but the logic is hard to follow. We can just give `authtype` a initial value.

Comment (Evengard) [reply src/Cedar/Proto_PPP.c:1281]: I considered doing that, but I couldn't select one tbh. I'd leave this and add a default value, just which one?

File: src/Cedar/Proto_PPP.c (hunk 1 above)
Comment (Evengard) [reply src/Cedar/Proto_PPP.c:2326]: You're actually right, that indeed means the packet was processed, and if the pipe is broken that doesn't change the fact. Probably the reason I did it in the first place.

File: src/Cedar/Proto_PPP.c (hunk 3 above)
Comment (Evengard) [reply src/Cedar/Proto_PPP.c:1281]: Reason why I'd leave it - if the client for some weird reason resends an identity packet, better to reset it here anyway.

**Review by domosekai — APPROVED** [review]
